nvm.sh

```diff
@@ -1839,7 +1839,12 @@ nvm_get_arch() {
   elif [ "_${NVM_OS}" = "_aix" ]; then
     HOST_ARCH=ppc64
   else
-    HOST_ARCH="$(command uname -m)"
```

Comment:
Seems like we should keep this line, so it can be referenced in the next line, and then overridden in the else branch

Reply:
Still working on this PR, there are some edge cases where NodeJs is compatible with arm64 in 16.0.0 and above versions. So I am planning to put conditions before downloading the NodeJs version artifacts not in checking architecture function. 

As of these changes, it will always download **x86_64** files even if there **arm64** file exists

Reply:
It'd be good to document the logic, either in comments or in the readme - eg, "there's node arm64 builds in these versions, but only these versions actually work on an m1, so we do X"

Reply:
Yeah, that would be great to mention

Reply:
to be clear tho, my comment meant "i only want to invoke uname once, in one code path" - currently the code has potentially two invocations of uname.

Reply:
Actually, I planned to revert those changes and add those conditions somewhere else. No offence but, that's the reason that I haven't responded to comment 😬